source/features/swap-branches-on-compare.tsx

```diff
@@ -5,7 +5,7 @@ import features from '../libs/features';
 import {getRepoPath, getRepoURL} from '../libs/utils';
 
 function init() {
-	const references = (getRepoPath() as string)
+	const references = getRepoPath()
```

Comment:
I’m surprised that this isn’t required anymore just because the type is `string?`. It could still be a type error at runtime… unless TS actually understands that it will never happen.

Reply:
Yeah, it's a good point. I'll see why it's not complaining.

Reply:
Which brings up a good point. We should guard against undefined. Instead of 

```
function init() {
	const references = getRepoPath()
		.replace('compare/', '')
		.split('...')
		.reverse();
...
```

we could simply do

```
function init() {
	const references = (getRepoPath() || '')
		.replace('compare/', '')
		.split('...')
		.reverse();
...
```

In fact the current code probably breaks since if no repo path is found.

```
false.replace('compare/', '')
VM118:1 Uncaught TypeError: false.replace is not a function
    at <anonymous>:1:7
```

Reply:
`(getRepoPath() || '')` would be fine but that would never happen because this feature is restricted to run only on repo pages, so `getRepoPath` will always return a string.